Snowflake.Data.Tests/IntegrationTests/SFConnectionWithTomlIT.cs

```diff
@@ -0,0 +1,135 @@
+    {
+        private static readonly SFLogger s_logger = SFLoggerFactory.GetLogger<SFConnectionIT>();
+
+        private static string s_workingDirectory;
+
+
+        [SetUp]
+        public new void BeforeTest()
+        {
+            s_workingDirectory ??= Path.Combine(TestContext.CurrentContext.WorkDirectory, "../../..", $"toml_config_folder");
+            if (!Directory.Exists(s_workingDirectory))
+            {
+                Directory.CreateDirectory(s_workingDirectory);
+            }
+            var snowflakeHome = Environment.GetEnvironmentVariable(TomlConnectionBuilder.SnowflakeHome);
+            Environment.SetEnvironmentVariable(TomlConnectionBuilder.SnowflakeHome, s_workingDirectory);
+        }
+
+        [TearDown]
+        public new void AfterTest()
+        {
+            Directory.Delete(s_workingDirectory, true);
+        }
+
+        [Test]
+        public void TestLocalDefaultConnectStringReadFromToml()
+        {
+            var snowflakeHome = Environment.GetEnvironmentVariable(TomlConnectionBuilder.SnowflakeHome);
+            CreateTomlConfigBaseOnConnectionString(ConnectionString);
+            Environment.SetEnvironmentVariable(TomlConnectionBuilder.SnowflakeHome, s_workingDirectory);
+            try
+            {
+                using (var conn = new SnowflakeDbConnection())
+                {
+                    conn.Open();
+                    Assert.AreEqual(ConnectionState.Open, conn.State);
+                }
+            }
+            finally
+            {
+                Environment.SetEnvironmentVariable(TomlConnectionBuilder.SnowflakeHome, snowflakeHome);
+            }
+        }
+
+        [Test]
+        public void TestThrowExceptionIfTomlNotFoundWithOtherConnectionString()
+        {
+            var snowflakeHome = Environment.GetEnvironmentVariable(TomlConnectionBuilder.SnowflakeHome);
+            var connectionName = Environment.GetEnvironmentVariable(TomlConnectionBuilder.SnowflakeDefaultConnectionName);
+            CreateTomlConfigBaseOnConnectionString(ConnectionString);
+            Environment.SetEnvironmentVariable(TomlConnectionBuilder.SnowflakeHome, s_workingDirectory);
+            Environment.SetEnvironmentVariable(TomlConnectionBuilder.SnowflakeDefaultConnectionName, "notfoundconnection");
+            try
+            {
+                using (var conn = new SnowflakeDbConnection())
+                {
+                    Assert.Throws<SnowflakeDbException>(() => conn.Open(), "Unable to connect. Specified connection name does not exist in connections.toml");
+                }
+            }
+            finally
+            {
+                Environment.SetEnvironmentVariable(TomlConnectionBuilder.SnowflakeHome, snowflakeHome);
+                Environment.SetEnvironmentVariable(TomlConnectionBuilder.SnowflakeDefaultConnectionName, connectionName);
+            }
+        }
+
+        [Test]
+        public void TestThrowExceptionIfTomlFromNotFoundFromDbConnection()
```

Comment:
when I run this test on its own it is failing.
If you would set SNOWFLAKE_HOME to point to a directory where there is no toml file it would be understandable that the connection should fail to open.
But here I don't see the reason why opening connection should fail since the toml file is created at the beginning of tests.